src/script/backup/BackupRepository.ts

```diff
@@ -140,61 +117,36 @@ export class BackupRepository {
     }
   }
 
-  private async _exportHistory(progressCallback: ProgressCallback) {
-    const [conversationTable, eventsTable, usersTable] = this.backupService.getTables();
-    const tableData: Record<string, any[]> = {};
-
-    function streamProgress<T>(dataProcessor: (data: T[]) => T[]) {
-      return (data: T[]) => {
-        progressCallback(data.length);
-        return dataProcessor(data);
-      };
-    }
-
-    const conversationsData = await this.exportTable(conversationTable, streamProgress(preprocessConversations));
-    tableData[StorageSchemata.OBJECT_STORE.CONVERSATIONS] = conversationsData;
-
-    const eventsData = await this.exportTable(eventsTable, streamProgress(preprocessEvents));
-    tableData[StorageSchemata.OBJECT_STORE.EVENTS] = eventsData;
-
-    const usersData = await this.exportTable(usersTable, streamProgress(preprocessUsers));
-    tableData[StorageSchemata.OBJECT_STORE.USERS] = usersData;
-
-    return tableData;
-  }
-
-  private async exportTable<T>(table: Dexie.Table<T, unknown>, preprocessor: (tableRows: T[]) => T[]): Promise<any[]> {
-    const tableData: T[] = [];
-
-    await this.backupService.exportTable(table, tableRows => {
-      if (this.canceled) {
-        throw new CancelError();
-      }
-      const processedData = preprocessor(tableRows);
-      tableData.push(...processedData);
-    });
-    return tableData;
-  }
   private async compressHistoryFiles(
     user: User,
-    clientId: string,
-    exportedData: Record<string, any>,
+    clientId: string, // TODO: Add clientId to metadata
+    exportedData: Int8Array | Record<string, any[]>,
     password: string,
   ): Promise<Blob> {
-    const metaData = this.createMetaData(user, clientId);
-
     const files: Record<string, Uint8Array> = {};
 
-    const stringifiedMetadata = JSON.stringify(metaData, null, 2);
-    const encodedMetadata = new TextEncoder().encode(stringifiedMetadata);
+    if (this.canceled) {
+      throw new CancelError();
+    }
+
+    // If the exported data is an Int8Array, it is a Multiplatform backup
```

Comment:
can we align the naming within the codebase
```suggestion
    // If the exported data is an Int8Array, it is a cross-platform backup
```